src/libraries/System.Net.Primitives/src/System/Net/IPNetwork.cs

```diff
@@ -0,0 +1,400 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+#pragma warning disable SA1648 // TODO: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3595
+
+namespace System.Net
+{
+    /// <summary>
+    /// Represents an IP network in CIDR notation with a <see cref="IPAddress"/> and a prefix length.
```

Comment:
What does it mean for a type to represent something "in notation"?  It's parse and format methods support CIDR notation, but the type itself just represents an IP network, e.g. you can construct one from an IPAddress and an int prefix length and call Contains, and there's no notation in sight. :)

Reply:
The alternative representation is a subnet mask. E.g. 192.168.1.0/24 -> network 192.168.1.0 and subnet mask 255.255.255.0.

I guess the type could be notation neutral if it had constructors and properties for the subnet mask as well. There wouldn't be a parsable/tostring format for that though.

Reply:
> I guess the type could be notation neutral if it had constructors and properties for the subnet mask as well.

@Tratcher have you seen such feature requests popping up against your type? I'm skeptical if it's worth to make it notation neutral.

I wanted to make a reference to term "CIDR notation" in the docs, because I thought it helps to disambiguate how this type works semantically, even though I didn't mean that we represent it "in CIDR notation" in lexical means. However, since we disallow arbitrary CIDR pairs, this might be misleading. I will remove the reference to CIDR.

Reply:
No, but our use case is very specific. I don't expect this to be a blocker right now, it could be added later if there's interest.